### DISCUSSION TITLE
Embeed EPG in IPTV Playlist

### DISCUSSION THREAD

**Discussion by SharyRajpoot** [iptv-org/community#175] [Ideas]
I was working with one of my IPTV projects and was looking for a source of EPG for IPTV but all I found is if you want to get a specific channel EPG then **tvg-id** is which used for the EPG guides.
But on some EPG list and your EPG list, I noticed different **tvg-id** so this can make conflict if I used a different IPTV playlist and if it does not match or find EPG with provided id then this can say no EPG found...
To clean my discussion more easily here is an example with the same channel but different **tvg-id** 

This is from your iptv playlist which is provided [here](https://github.com/iptv-org/iptv)...
```
#EXTINF:-1 tvg-id="AnimaxJapan.jp" tvg-country="JP" tvg-language="Japanese;Vietnamese" tvg-logo="https://i.imgur.com/zpi6mQ3.png"
https://livecdn.fptplay.net/hda3/animaxport_hls.smil/playlist.m3u8
```

This is the same as above but different **tvg-id** from another source or server you can say.
```
#EXTINF:-1 tvg-id="Animax.jp" tvg-country="JP" tvg-language="...." tvg-logo="https:/......."
https://........
```

Now noticed here your **tvg-id** is different if we look for EGP in some other places like one I found which is I am using it as an example is this [Animax EPG](https://epg.best/available-channels?term=animax) here they are using **tvg-id=Animax.jp** to look for EPG guides but on your IPTV Playlist we are using **tvg-id="AnimaxJapan.jp"** which is different and can cause not found...

So now coming to the main point...
This is our playlist so why not we make a few bit of changes in the IPTV Playlist Standard Format.
Here now if we add **tvg-epg** tag inside the IPTV playlist then everything can be on the same page and we don't have to look another place for EPG if it's available then it's inside with the channel details if not then it's not available...

Example:
**Standard IPTV Format**
```
#EXTINF:-1 tvg-id="AnimaxJapan.jp" tvg-country="JP" tvg-language="Japanese;Vietnamese" tvg-logo="https://i.imgur.com/zpi6mQ3.png"
https://livecdn.fptplay.net/hda3/animaxport_hls.smil/playlist.m3u8
```

**Updated IPTV Format**
```
#EXTINF:-1 tvg-id="AnimaxJapan.jp" tvg-country="JP" tvg-language="Japanese;Vietnamese" tvg-logo="https://i.imgur.com/zpi6mQ3.png" tvg-epg="https://iptv-org.github.io/epg/guides/id/mncvision.id.epg.xml"
https://livecdn.fptplay.net/hda3/animaxport_hls.smil/playlist.m3u8
```

Let me know what you all think about this idea...
As it provides ease all in one place.

**Comment by freearhey** [comment]
I like this idea, but I have a concern that files will become very difficult to browse and maintain when each link has this attribute. For example here are just 3 links with new attribute:

```
#EXTINF:-1 tvg-id="AnimaxJapan.jp" tvg-country="JP" tvg-language="Japanese;Vietnamese" tvg-logo="https://i.imgur.com/zpi6mQ3.png" user-agent="Mozilla/5.0 Macintosh; Intel Mac OS X 10_14_5 AppleWebKit/537.36 KHTML, like Gecko Chrome/76.0.3809.25 Safari/537.36" tvg-url="https://iptv-org.github.io/epg/guides/id/mncvision.id.epg.xml" group-title="Animation",Animax Japan (Vietnamese Subs) (720p)
#EXTVLCOPT:http-user-agent=Mozilla/5.0 Macintosh; Intel Mac OS X 10_14_5 AppleWebKit/537.36 KHTML, like Gecko Chrome/76.0.3809.25 Safari/537.36
https://livecdn.fptplay.net/hda3/animaxport_hls.smil/playlist.m3u8
#EXTINF:-1 tvg-id="AnimaxJapan.jp" tvg-country="JP" tvg-language="Japanese;Vietnamese" tvg-logo="https://i.imgur.com/zpi6mQ3.png" user-agent="Mozilla/5.0 Macintosh; Intel Mac OS X 10_14_5 AppleWebKit/537.36 KHTML, like Gecko Chrome/76.0.3809.25 Safari/537.36" tvg-url="https://iptv-org.github.io/epg/guides/id/mncvision.id.epg.xml" group-title="Animation",Animax Japan (Vietnamese Subs) (720p) [Not 24/7]
#EXTVLCOPT:http-user-agent=Mozilla/5.0 Macintosh; Intel Mac OS X 10_14_5 AppleWebKit/537.36 KHTML, like Gecko Chrome/76.0.3809.25 Safari/537.36
https://livecdn.fptplay.net/hda3/animaxport_2000.stream/playlist.m3u8
#EXTINF:-1 tvg-id="AnimaxKorea.jp" tvg-country="KR" tvg-language="Korean" tvg-logo="https://i.imgur.com/zpi6mQ3.png" tvg-url="https://iptv-org.github.io/epg/guides/id/mncvision.id.epg.xml" group-title="Animation",Animax Korea [Offline]
http://132.145.83.248:9981/stream/channelid/343601055
```

Also, I would change the attribute name to `tvg-url` since "g" already implies "guide" as I understand it.
